### PR TITLE
computes the service-id using compute-descriptor function in consent handler

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- computes the service-id using compute-descriptor function in consent handler

## Why are we making these changes?

The consent handler incorrectly computes service-IDs when the `compute-descriptor` function dynamically adds parameters to the service description (e.g. when service-mapping is exclusive).

### CODE REVIEW

File: waiter/src/waiter/handler.clj [hunk 1]
```diff
@@ -1062,9 +1060,7 @@
       (when-not (seq service-description-template)
         (throw (ex-info "Unable to load description for token" {:log-level :info :status http-404-not-found :token token})))
       (let [auth-user (:authorization/user request)
-            service-id (-> service-description-template
-                           (sd/assoc-run-as-requester-fields auth-user)
-                           service-description->service-id)
+            service-id (request->consent-service-id request)
```
Comment (DaoWen) [inline waiter/src/waiter/handler.clj:1063]: Why is service-id computed separately so many places???

File: waiter/src/waiter/core.clj [hunk 2]
```diff
@@ -1067,6 +1067,14 @@
    :refresh-service-descriptions-fn (pc/fnk [[:state kv-store]]
                                       (fn refresh-service-descriptions-fn [service-ids]
                                         (sd/refresh-service-descriptions kv-store service-ids)))
+   :request->consent-service-id (pc/fnk [[:state kv-store service-description-builder service-id-prefix waiter-hostnames]
```
Comment (DaoWen) [inline waiter/src/waiter/core.clj:1070]: Why are we defining a separate function for the *consent* service-id? Shouldn't we just have one consistent way to compute the service-id? It seems like maintaining separate functions is setting us up for more inconsistency in the future.

**Review by DaoWen — APPROVED** [review]
I'm merging because this still puts us in a better situation that we're in now (bug fixed) — but we should probably address the underlying problem of computing the service-id in multiple places without a single consistent function.

File: waiter/src/waiter/core.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/src/waiter/core.clj:1070]: This gap is reduced in the new way, as now we use the common `compute-descriptor` with an override for allowing run-as-requester without permission (the consent workflow is supposed to give us the permission) to get the service ID.

File: waiter/src/waiter/handler.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/src/waiter/handler.clj:1063]: The two places it is computed in the consent worflow:

1. Computed to include the HTML form we generate to render the consent form
2. Computed to confirm that the service-ID sent in the form submission matches the service we are trying to launch.

**Pull request merged**